R/vetiver-model.R

```diff
@@ -47,13 +48,22 @@ vetiver_model <- function(model,
                           ...,
                           description = NULL,
                           metadata = list(),
-                          save_ptype = TRUE,
+                          save_prototype = TRUE,
+                          save_ptype = deprecated(),
                           versioned = NULL) {
 
+    if (lifecycle::is_present(save_ptype)) {
+        lifecycle::deprecate_soft(
+            "0.2.0",
+            "vetiver_model(save_ptype)",
+            "vetiver_model(save_prototype)"
+        )
+        save_prototype <- save_ptype
+    }
     if (is_null(description)) {
         description <- vetiver_create_description(model)
     }
-    ptype <- vetiver_create_ptype(model, save_ptype, ...)
+    ptype <- vetiver_create_ptype(model, save_prototype, ...)
```

Comment:
Do we still want to call the attribute `ptype`?

Reply:
My concern about changing that is we would need to set up some handling to read models pinned with the old name. If we keep the name the same as it is now, we can continue to read models pinned with old versions of vetiver. What do you think?

Reply:
It feels a bit strange that while the argument is called `save_prototype`, it doesn't save `prototype`, it saves `ptype`. I don't think it's too much of a mental leap, but worth noting. Overall, I don't think it is as big of a concern as the backwards compatibility/maintainability, so I'm good with it staying ptype 👍

Reply:
OK, it's actually way simpler to look for either `ptype` or `prototype` than I was imagining; see 9d2076ad2c2bd70f939907da3591a63678390bff.

With this installed I can read old pinned models just fine:

``` r
library(vetiver)
library(pins)

b <- board_rsconnect()
#> Connecting to RSC 2022.11.0 at <https://colorado.rstudio.com/rsc>
v <- b %>% vetiver_pin_read("julia.silge/wow-another-mtcars-model")
v
#> 
#> ── julia.silge/wow-another-mtcars-model ─ <butchered_lm> model for deployment 
#> An OLS linear regression model using 10 features
```

<sup>Created on 2022-12-09 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

I'm going to change what the internal element is called. 👍

Reply:
I ended up changing more than I initially thought I would, but I believe this is fairly future-proof and won't cause unexpected problems.